source/renderer/app/components/wallet/send-form/WalletSendAssetsConfirmationDialog.js

```diff
@@ -260,43 +157,61 @@ export default class WalletSendAssetsConfirmationDialog extends Component<
     (this.props.isHardwareWallet || this.form.$('passphrase').isValid) &&
     submitOnEnter(this.submit, event);
 
-  renderConfirmationElement = (isHardwareWallet: boolean) => {
+  renderConfirmationElement = (
+    isHardwareWallet: boolean
+  ): React$Element<*> | null => {
     const passphraseField = this.form.$('passphrase');
     const { areTermsAccepted } = this.state;
     const {
       hwDeviceStatus,
       isFlight,
+      totalAmount,
       onExternalLinkClick,
-      walletName,
+      wallet,
       isTrezor,
     } = this.props;
 
+    let returnJSX = null;
     if (!isFlight || (isFlight && areTermsAccepted)) {
-      if (isHardwareWallet) {
-        return (
+      const walletWarning = (
+        <div className={styles.flightCandidateWarning}>
+          <FormattedHTMLMessage
+            {...getMessages().emptyingWarning}
+            tagName="p"
+          />
+        </div>
+      );
+
+      returnJSX = isHardwareWallet ? (
+        <>
           <div className={styles.hardwareWalletStatusWrapper}>
             <HardwareWalletStatus
               hwDeviceStatus={hwDeviceStatus}
-              walletName={walletName}
+              walletName={wallet?.name}
```

Comment:
@danielmain please reference all needed wallet props around line 172.
e.g. `const { name, amount } = wallet;`

Reply:
As you like, both are valid options to access securely a property in an object just **Object destructuring** sacrifices another line and has no added value against **optional chaining**.

Reply:
I know but I would like to keep this part uniform (for now). Thanks!